controller/backing_image_data_source_controller.go

```diff
@@ -927,10 +927,21 @@ func (c *BackingImageDataSourceController) stopMonitoring(bidsName string) {
 		log.Warn("No monitor goroutine for stopping")
 		return
 	}
-	log.Infof("Stopping monitoring")
-	close(stopCh)
+	log.Info("Stopping monitoring")
+
+	// https://github.com/golang/go/blob/master/src/runtime/chan.go
+	// If the channel is closed and buffer is empty, the chanrecv() will return (true, false)
+	select {
```

Comment:
This workaround is OK, but why not just send to a signal to the stop channel in the sync method when the data source is not responsible, then leave the stopMonitor help with closing channel instead?

The concept is `who create the resource, then she needs to handle it like pointer`, so the ownership will be clear.

Reply:
Yes, I think you're right, it should just let `stop Monitoring()` do all about stopping the monitor. So I send the signal `struct{}{}` to the stop channel instead.